account_invoice_fixed_discount/models/account_invoice.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+import odoo.addons.decimal_precision as dp
+from odoo.exceptions import ValidationError
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    @api.multi
+    def get_taxes_values(self):
+        self.ensure_one()
+        vals = {}
+        for line in self.invoice_line_ids.filtered('discount_fixed'):
+            vals[line] = {
+                'price_unit': line.price_unit,
+                'discount_fixed': line.discount_fixed,
+            }
+            price_unit = line.price_unit - line.discount_fixed
+            line.update({
+                'price_unit': price_unit,
+                'discount_fixed': 0.0,
+            })
+        tax_grouped = super(AccountInvoice, self).get_taxes_values()
+        for line in vals.keys():
+            line.update(vals[line])
+        return tax_grouped
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.invoice.line"
+
+    discount_fixed = fields.Float(
+        string="Discount (Fixed)",
+        digits=dp.get_precision('Product Price'),
+        help="Fixed amount discount.")
+
+    @api.onchange('discount')
```

Comment:
Why not leaving both fields, why this restriction, logically you can benefit on both into one promotion, let's say..., it will not breake anything, just when you compute the price to take the discount in calculation

Reply:
To not overcomplicate it and to add the fixed_discount with the least impact/extensions on the standard as it was just what I needed.

Reply:
@lreficent Ok, but can you please added in readme roadmap, to know about it..

Reply:
yes, done!